airbyte-webapp/src/views/Connector/ServiceForm/useAuthentication.tsx

```diff
@@ -0,0 +1,210 @@
+  );
+  // Get the part of the spec that `rootPath` point to
+  const credentialsSpecRoot = getIn(spec, specPath) as JSONSchema7 | undefined;
+
+  if (!credentialsSpecRoot?.properties) {
+    // if the part doesn't exist in the spec (which should not happen) we just show the auth button always.
+    return true;
+  }
+
+  // To find the value we're expecting, we run through all properties inside that matching spec inside the conditional
+  // to find the one that has a `const` value in it, since this is the actual value that will be written into the conditional
+  // field itself once it's selected.
+  const constProperty = Object.entries(credentialsSpecRoot.properties)
+    .map(([key, prop]) => [key, typeof prop !== "boolean" ? prop.const : undefined] as const)
+    .find(([, constValue]) => !!constValue);
+
+  // If none of the conditional properties is a const value, we'll also show the auth button always (should not happen)
+  if (!constProperty) {
+    return true;
+  }
+
+  // Check if the value in the form match the found `const` value from the spec. If so we know the conditional
+  // is on the right option.
+  const [key, constValue] = constProperty;
+  const value = getIn(values, makeConnectionConfigurationPath(stripNumericalEntries([...rootPath, key])));
+  return value === constValue;
+};
+
+interface AuthenticationHook {
+  /**
+   * Returns whether a given field path should be hidden, because it's part of the
+   * OAuth flow and will be filled in by that.
+   */
+  isHiddenAuthField: (fieldPath: string) => boolean;
+  /**
+   * A record of all formik errors in hidden authentication fields. The key will be the
+   * name of the field and the value an error code. If no error is present in a field
+   * it will be missing from this object.
+   */
+  hiddenAuthFieldErrors: Record<string, string>;
+  /**
+   * This will return true if the auth button should be visible and rendered in the place of
+   * the passed in field, and false otherwise.
+   */
+  shouldShowAuthButton: (fieldPath: string) => boolean;
+}
+
+export const useAuthentication = (): AuthenticationHook => {
+  const { values, getFieldMeta, submitCount } = useFormikContext<ServiceFormValues>();
+  const { selectedConnector } = useServiceForm();
+
+  const allowOAuthConnector = useFeature(FeatureItem.AllowOAuthConnector);
+
+  const advancedAuth = selectedConnector?.advancedAuth;
+  const legacyOauthSpec = selectedConnector?.authSpecification?.oauth2Specification;
+
+  const spec = selectedConnector?.connectionSpecification as JSONSchema7;
+
+  const isAuthButtonVisible = useMemo(() => {
+    const shouldShowAdvancedAuth =
+      advancedAuth && shouldShowButtonForAdvancedAuth(advancedAuth.predicateKey, advancedAuth.predicateValue, values);
+    const shouldShowLegacyAuth =
+      legacyOauthSpec && shouldShowButtonForLegacyAuth(spec, legacyOauthSpec.rootObject as Path, values);
+    return Boolean(allowOAuthConnector && (shouldShowAdvancedAuth || shouldShowLegacyAuth));
```

Comment:
Why wrap this response in a `Boolean()`?

Reply:
`shouldShowAdvancedAuth` and `shouldShowLegacyAuth` are of type `boolean | undefined`, since our check above is e.g. `advancedAuth && shouldShow...`, meaning if `advancedAuth` is not set in the spec and `undefined` `shouldShowAdvancedAuth` will be undefined and not `false`, since the value is simply assigned from that variable (and the rest behind `&&` would only be executed if the variable before is truthy.

So without that `Boolean` our result here would be `boolean | undefined` again, since in case `allowOAuthConnector` is `true`, and `shouldShowAdvancedAuth` is `undefined` the statement will end there and resolve to `undefined` itself again. Since we want to have `false` in this case, we just cast it to boolean manually, so the `undefined` will become `false`.